### ISSUE TITLE
Add a Switch to Numeric EditTextPreferences to activate/deactivate them

### ISSUE THREAD

**Issue by GrazianoCapelli** [BasicAirData/GPSLogger#164]
Enabling and Disabling _Distance Filter_, _Interval Filter_, and _Manual Correction_ is not comfortable.
When I want to disable one of these preferences, I have to open the dialog, move the cursor to the right position of the EditText, set it to zero and click OK; when I later want to re-enable it, I have to remember the old value (:unamused:), open the dialog, set it, and click OK.

We could add a switch to the preference in order to simplify this process:

![manual_correction](https://user-images.githubusercontent.com/6907566/152942092-712f3121-75b1-4356-9b01-ead0896a311d.jpg)

**Comment by VasireddyGanesh** [comment]
@GrazianoCapelli  Can you please answer the following questions to help me understand the requirements properly?

1. Do I need to add a switch button along with the existing numeric EditTextPreference ?

2. Do I need to add a switch button for all three preferences (Distance Filter, Interval Filter, and Manual Correction) ?

**Comment by GrazianoCapelli** [comment]
@VasireddyGanesh

>     1. Do I need to add a switch button along with the existing numeric EditTextPreference ?

Yes, the preference should keep working as edittext, but we would like to add a switch on side in order to activate/deactivate the preference (with the value specified by the user).
 
>     2. Do I need to add a switch button for all three preferences (Distance Filter, Interval Filter, and Manual Correction) ?

Yes, for all 3 preferences.

**Comment by VasireddyGanesh** [comment]
> but we would like to add a switch on side in order to activate/deactivate the preference (with the value specified by the user).

@GrazianoCapelli Should I create a custom preference that combines an EditText and a Switch, or would it be sufficient to have two separate preferences, one for the EditText and another for the Switch ?

**Comment by GrazianoCapelli** [comment]
@VasireddyGanesh we should implement a custom preference that combines an EditText and a Switch.
The same custom preference layout can be used for all the 3 preferences.
